test/e2e/elasticsearch_test.go

```diff
@@ -93,7 +94,41 @@ func (suite *ElasticSearchTestSuite) TestSimpleProd() {
 	ProductionSmokeTest("simple-prod")
 }
 
+func (suite *ElasticSearchTestSuite) TestEsIndexPrefixes() {
```

Comment:
This should be renamed `TestEsIndexCleanerWithIndexPrefix`.

Reply:
@objectiser We already had a test case which covered the index cleaner:  https://github.com/jaegertracing/jaeger-operator/blob/master/test/e2e/elasticsearch_test.go#L96-L144  

I had originally only tested prefixes in the new test case, but when I asked @pavolloffay a question about it in email he replied "When the prefix is specified the index cleaner should pick it up and use the prefix too. The #606 should ideally also deploy index cleaner and verify it works as expected" so I had it deploy the index cleaner as well.

I can change the test name or add comments if you wish.

Reply:
Yeah that sounds reasonable.